engine/src/gfx/cockpit_gfx_utils.cpp

```diff
@@ -0,0 +1,255 @@
+    const float inner_distance = end_reticle + (start_reticle - end_reticle) * lock_percent;
+    const float outer_distance = inner_distance + bracket_size;
+    const float bracket_distance = (bracket_inner_or_outer ? inner_distance : outer_distance);
+
+    const float adjusted_inner_distance = inner_distance * bounded_r_size;
+    const float adjusted_outer_distance = outer_distance * bounded_r_size;
+    const float ajusted_bracket_distance = bracket_distance * bounded_r_size;
+    const float ajusted_bracket_width = bracket_width * bounded_r_size;
+
+
+    static VertexBuilder<> verts;
+    verts.clear();
+
+    verts.insert( location + cam_p * adjusted_inner_distance );
+    verts.insert( location + cam_p * adjusted_outer_distance );
+
+    verts.insert( location + cam_p * ajusted_bracket_distance + cam_q * ajusted_bracket_width );
+    verts.insert( location + cam_p * ajusted_bracket_distance - cam_q * ajusted_bracket_width );
+
+    verts.insert( location - cam_p * adjusted_inner_distance );
+    verts.insert( location - cam_p * adjusted_outer_distance );
+
+    verts.insert( location - cam_p * ajusted_bracket_distance + cam_q * ajusted_bracket_width );
+    verts.insert( location - cam_p * ajusted_bracket_distance-cam_q * ajusted_bracket_width );
+
+    verts.insert( location + cam_q * adjusted_inner_distance );
+    verts.insert( location + cam_q * adjusted_outer_distance );
+
+    verts.insert( location + cam_q * ajusted_bracket_distance + cam_p * ajusted_bracket_width );
+    verts.insert( location + cam_q * ajusted_bracket_distance - cam_p * ajusted_bracket_width );
+
+    verts.insert( location - cam_q * adjusted_inner_distance );
+    verts.insert( location - cam_q * adjusted_outer_distance );
+
+    verts.insert( location - cam_q * ajusted_bracket_distance + cam_p * ajusted_bracket_width );
+    verts.insert( location-cam_q * ajusted_bracket_distance - cam_p * ajusted_bracket_width );
+
+    return verts;
+}
+
+/*
+ * The locking icon will grow and shrink as lock as achieved.
+ * The animated locking icon will also move from the reticle to the target and apparently rotate.
+ */
+VertexBuilder<> GetAnimatedLockingIcon(const QVector &location, const Vector& cam_p,
+                            const Vector& cam_q, const Vector& cam_r,
+                            const float& r_size, const float& lock_percent) {
+    static const float lock_line = GameConfig::GetVariable( "graphics", "hud", "LockConfirmLineLength", 1.5 );
+    static const float diamond_size = GameConfig::GetVariable( "graphics", "hud", "DiamondSize", 2.05 );
+    static const float theta_speed = GameConfig::GetVariable( "graphics", "hud", "DiamondRotationSpeed", 1.0 );
+
+    const float max = diamond_size * r_size * 0.75 * end_reticle;
+    const float coord = end_reticle+(start_reticle-end_reticle)*lock_percent;
+    const double rtot = 1./sqrtf( 2 );
+
+    //this causes the rotation!
+    const float theta = 4*M_PI*lock_percent*theta_speed;
```

Comment:
I'm having issues with `M_PI` not being defined, at least on Windows. The fix appears to be to add the following 3 lines at the top of the file:

```c++11
#undef __STRICT_ANSI__
#define _USE_MATH_DEFINES
#include <cmath>
```

Reply:
@stephengtuggy perhaps that should be wrapped by:
```
#ifdef WIN32
...
#endif // WIN32
```
http://www.cplusplus.com/forum/beginner/105176/

Or better yet - we should move the Boost version: https://www.boost.org/doc/libs/1_78_0/libs/math/doc/html/math_toolkit/constants.html